app/models/manageiq/providers/embedded_ansible/automation_manager/scm_credential.rb

```diff
@@ -1,37 +1,49 @@
 class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ScmCredential < ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Credential
-  COMMON_ATTRIBUTES = {
-    :userid   => {
-      :label     => N_('Username'),
-      :help_text => N_('Username for this credential')
+  COMMON_ATTRIBUTES = [
+    {
+      :component  => 'text-field',
+      :label      => N_('Username'),
+      :helperText => N_('Username for this credential'),
+      :name       => 'userid',
+      :id         => 'userid',
     },
-    :password => {
-      :type      => :password,
-      :label     => N_('Password'),
-      :help_text => N_('Password for this credential')
-    }
-  }.freeze
+    {
+      :component  => 'password-field',
+      :label      => N_('Password'),
+      :helperText => N_('Password for this credential'),
+      :name       => 'password',
+      :id         => 'password',
+      :type       => 'password',
+    },
+  ].freeze
 
   # rubocop:disable Layout/AlignHash
   #
   # looks better to align the nested keys to the same distance, instead of
   # scope just for the hash in question (which is what rubocop does.
-  EXTRA_ATTRIBUTES = {
-    :ssh_key_unlock => {
-      :type       => :password,
+  EXTRA_ATTRIBUTES = [
+    {
+      :component  => 'password-field',
       :label      => N_('Private key passphrase'),
-      :help_text  => N_('Passphrase to unlock SSH private key if encrypted'),
-      :max_length => 1024
+      :helperText => N_('Passphrase to unlock SSH private key if encrypted'),
+      :name       => 'options.ssh_key_unlock',
+      :id         => 'ssh_key_unlock',
+      :maxLength  => 1024,
+      :type       => 'password',
     },
-    :ssh_key_data => {
-      :type       => :password,
-      :multiline  => true,
-      :label      => N_('Private key'),
-      :help_text  => N_('RSA or DSA private key to be used instead of password')
-    }
-  }.freeze
+    {
```

Comment:
Let's swap the order of the key/passphrase here to be consistent with the others...